### PR TITLE
BUG: JSON serialization with orient split fails roundtrip with MultiIndex

### LINKED ISSUE #50456: BUG: JSON serialization with orient split fails roundtrip with MultiIndex
When saving a DataFrame to JSON with `orient='split'` and then loading it again, the loaded dataframe is different from the original if columns are a multiindex.

```python
>>> df = DataFrame([[1, 2], [3, 4]],
...                columns=pd.MultiIndex.from_arrays([["2022", "2022"], ['JAN', 'FEB']]))
>>> df
  2022    
   JAN FEB
0    1   2
1    3   4

>>> read_json(df.to_json(orient='split'), orient='split')
  2022 JAN
  2022 FEB
0    1   2
1    3   4
```

The problem seems to be that the JSON stores the format as `{"columns":[["2022","JAN"],["2022","FEB"]], ...}`, but when creating the loaded DataFrame the `columns` value is passes as that, and `DataFrame(data, columns=[["2022","JAN"],["2022","FEB"]])` produces the incorrect result.

We can fix this by either changing how data is stored in the JSON, or how the dataframe is created. Personally, I think it makes more sense to store the data in the JSON in the way expected by the dataframe constructor.

CC: @MarcoGorelli

### PR DESCRIPTION
**Pull request by labibdotc**
- Closes #50456
- xFail for mark split deprecated.
- All [code checks passed].
- Added an entry in the latest `doc/source/whatsnew/v2.0.0.rst`

### CODE REVIEW

File: pandas/tests/io/json/Untitled-1.ipynb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
```
Comment (mroeschke) [inline pandas/tests/io/json/Untitled-1.ipynb:1]: This should not be commited

File: pp.py [hunk 2]
```diff
@@ -0,0 +1,3 @@
+import pandas
```
Comment (mroeschke) [inline pp.py:1]: This should not be commited either

File: pandas/tests/io/json/test_pandas.py [hunk 3]
```diff
@@ -1528,13 +1528,6 @@ def test_timedelta_as_label(self, date_format, key):
             ("index", "{\"('a', 'b')\":{\"('c', 'd')\":1}}"),
             ("columns", "{\"('c', 'd')\":{\"('a', 'b')\":1}}"),
             # TODO: the below have separate encoding procedures
-            pytest.param(
-                "split",
-                "",
-                marks=pytest.mark.xfail(
-                    reason="Produces JSON but not in a consistent manner"
-                ),
-            ),
```
Comment (datapythonista) [inline pandas/tests/io/json/test_pandas.py:1537]: This is not the test, the test is `test_roundtrip_multiindex` and what needs to be removed is the `xfail` decorator.

**Review by datapythonista — COMMENTED** [review]
Thanks @labibdotc for the PR. Besides addressing the comments when you've got the time, can you also merge main into your branch and fix the conflicts. You may want to add your release note not at the very end of the list. That should help prevent conflicts.

**Comment by mroeschke** [comment]
Thanks for the pull request, but it appears to have gone stale. If interested in continuing, please merge in the main branch, address any review comments and/or failing tests, and we can reopen.

**Pull request closed without merging**